app/response/utils.py

```diff
@@ -217,6 +217,42 @@ def add_closing(request_id, reason_ids, email_content):
         _send_response_email(request_id, privacy, email_content)
 
 
+def add_reopened(request_id, date, tz_name, email_content):
+    """
+    Create and store a re-opened-determination for the specified request and update the request accordingly.
+
+    :param request_id: FOIL request ID
+    :param date: string of new date of request completion
+    :param tz_name: client's timezone name
+    :param email_content: email body associated with the reopened request
+
+    """
+    if Requests.query.filter_by(id=request_id).one().status == request_status.CLOSED:
+        new_due_date = process_due_date(datetime.strptime(date, '%Y-%m-%d'), tz_name)
+        update_object(
+            {'due_date': new_due_date},
+            Requests,
+            request_id
+        )
+        privacy = RELEASE_AND_PUBLIC
+        response = Determinations(
+            request_id,
+            privacy,
+            determination_type.REOPENED,
+            None,
+            new_due_date
+        )
+        create_object(response)
+        _create_response_event(response, event_type.REQ_REOPENED)
+        _send_response_email(request_id, privacy, email_content)
+        update_object(
+            {'status': request_status.IN_PROGRESS,
+             'agency_description_release_date': None},
+            Requests,
+            request_id
+        )
```

Comment:
It looks like it's possible to update the request once, please do that, and do it before you send the email.